### ISSUE TITLE
Naming Style: Underscore does not trigger violation in camel_case and pascal_case style

### ISSUE THREAD

**Issue by nphmuller** [dotnet/roslyn#22745]
**Version Used**: The version in VS15.4.0

**Steps to Reproduce**:

Git clone the following repo: https://github.com/nphmuller/RoslynNameStyleUnderscore.git

**OR**

1. Create the following .editorconfig rule:
```
dotnet_naming_rule.private_fields_are_camel_case.symbols = private_field
dotnet_naming_rule.private_fields_are_camel_case.style = camel_case
dotnet_naming_rule.private_fields_are_camel_case.severity = warning

dotnet_naming_symbols.private_field.applicable_accessibilities = private
dotnet_naming_symbols.private_field.applicable_kinds = field

dotnet_naming_style.camel_case.capitalization = camel_case
dotnet_naming_style.pascal_case.capitalization = pascal_case
```
2. Insert the following code snippet:
``` c#
    public class Class1
    {
        private string camelCase; // (1)
        private string PascalCase;  // (2)

        private string _camelCase;  // (3)
        private string _PascalCase;  // (4)
    }
```

**Expected Behavior**:
Field 2, 3 and 4 should throw a warning.

**Actual Behavior**:
Field 2 throws a warning.

# Additionally
Change line 2 from the editorconfig file from `camel_case` to `pascal_case`. 

**Expected Behavior**:
Field 1, 3 and 4 should throw a warning.

**Actual Behavior**:
Field 1 throws a warning.

**Comment by sharwell** [comment]
Case 4 is a duplicate of #18354.

💭 Cases 3 and 6 seem to be a failure of the IDE to know what PascalCase is.

**Comment by jinujoseph** [comment]
Reported at [DC](https://developercommunity.visualstudio.com/content/problem/588118/naming-violation-for-all-caps-suggests-wrong-fix-o.html)

**Comment by WPJorge** [comment]
@akhera99 is there any update on this topic?
